lenskit-eval/src/main/java/org/lenskit/eval/traintest/recommend/RecommendEvalTask.java

```diff
@@ -303,13 +305,17 @@ public ConditionEvaluator createConditionEvaluator(AlgorithmInstance algorithm,
             return null;
         }
 
+        boolean useDetails = tlb != null; // writing requires details
```

Comment:
`tlb` (table layout builder) is null when we are are not writing the recommendations to a file; it is non-null when we are (and therefore require details so we can get a score).